packages/data-store/src/migration/v2/v1-data-store.ts

```diff
@@ -0,0 +1,35 @@
+export type LocalStorage = {
+  getItem: (key: string) => Promise<string | null>;
+  setItem: (key: string, value: string) => Promise<void>;
+  removeItem: (key: string) => Promise<void>;
+};
+
+let _localStorage: LocalStorage;
+
+/**
+ * Legacy data-store was based on AsyncLocalStorage
+ * The wallet will inject its implementation here, so that the migration scripts can use it
+ *
+ * @param _impl
+ */
+export function setV1LocalStorage(_impl: LocalStorage) {
+  _localStorage = _impl;
```

Comment:
could `_impl` be null or undefined here?

Reply:
fixed